contracts/UniswapV2Library.sol

```diff
@@ -20,7 +22,7 @@ contract UniswapV2Library is IUniswapV2Library {
     }
 
     // calculates the CREATE2 address for a pair without making any external calls
-    function pairFor(address tokenA, address tokenB) internal pure returns (address pair) {
+    function pairFor(address tokenA, address tokenB) internal view returns (address pair) {
```

Comment:
Reads the immutable factory? Seems like a solidity bug

Reply:
Ah, I missed the `factory` read.  I remember this coming up in discussion on Solidity mailing list.  I'm sad to see they went down that path.  😢  Sadly, it isn't a bug, it is a design decision.

Reply:
@moodysalem is correct. i was annoyed by this, seems like a bug to me but guess i could see why others might disagree.

something to note is that calling `pairFor` doesn't work from inside constructors of contracts inheriting from UniswapV2Library, because of the immutable factory dependency, so maybe there's some rule about being able to call pure functions inside of constructors at play here.